src/error.rs

```diff
@@ -29,6 +29,7 @@ pub enum Error {
     CannotParse(Type, Vec<u8>),
     IoError(::std::io::Error),
     TrailingData,
+    InvalidIndefiniteString,
```

Comment:
I'm not sure what we want to do with the cargo version. I added a new enum variant to this error which according to the rust guide would need a new major version change.

I also left the `IndefiniteLenNotSupported` since it's still returned in several parts of code (`uint`/`nint`/`tag`), but those are not possible in the cbor rfc. It only mentions array/map/strings as being able to be indefinite encoded. Maybe we should rename it instead then to reflect it being invalid structure rather than not being supported by the library?